lib/preprocessor.js

```diff
@@ -42,47 +48,48 @@ module.exports.run = function r(contract) {
 
   while (keepRunning) {
     try {
-      const ast = SolidityParser.parse(contract);
+      const ast = SolidityParser.parse(contract, { range: true });
       keepRunning = false;
-      SolExplore.traverse(ast, {
-        enter(node, parent) { // eslint-disable-line no-loop-func
-          // If consequents
-          if (node.type === 'IfStatement' && node.consequent.type !== 'BlockStatement') {
-            contract = blockWrap(contract, node.consequent);
+      blocksToWrap = [];
+      SolidityParser.visit(ast, {
```

Comment:
Hi @area , could you also remove the while loop in this PR?
I have made a PR to do that but in the old and forked version: https://github.com/leapdao/solidity-coverage/pull/3/files
It relies on the assumption that nodes are visited in order as in the contract string.

Reply:
Now done, and though I didn't end up using your code, thank you for drawing attention to the issue!